src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -222,6 +222,7 @@ public void UpdatePivot()
                 {
                     codesToQuery.Add(codeHash);
                 }
+                codesToQuery.Sort();
```

Comment:
Not exactly about this line but we create a List and then do ToArray, can we avoid that? I see that as a pattern in snap sync, I think this could optimize memory and GC a lot.

Can `request.CodesRequest` be `IReadOnlyList`? Also maybe insertion sort here?

Reply:
Reusing the Lists now; creating the arrays for now is safer and will need a deeper dive following the concurrency and lifetimes of the arrays to be able to reuse

Lists can be 1k+ so insertion sort would be slow?